app/main.py

```diff
@@ -1,3 +1,70 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        elif isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
+            return NotImplemented
```

Comment:
Good approach, but ir's actually implemented, so it should rise an Error. Also you don't need else here, since you will anyhow come down to this. Consider just saving common python logic inside your classes, to have an expected behavior:
```
3 + 4 == 7
unsupported operand type(s) for +: 'int' and 'str'
Think a little about it before implementing.
```